.github/workflows/app-integration.yaml

```diff
@@ -0,0 +1,58 @@
+name: app
+on:
+  - merge_group
+  - push
+  - pull_request
+
+# Only build for latest push/PR unless it's main or release/
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' && !startsWith( github.ref, 'refs/heads/release/' ) && !startsWith( github.ref, 'refs/heads/gh-readonly-queue/' ) }}
+
+defaults:
+  run:
+    shell: bash
+
+permissions: {}
+
+jobs:
+  integration-tests:
+    strategy:
+      matrix:
+        debian_version:
+          - bookworm
+    runs-on: ubuntu-latest
+    container: debian:${{ matrix.debian_version }}
+    services:
+      httpbin:
+        image: kennethreitz/httpbin # zizmor: ignore[unpinned-images]
+        ports:
+          - 80:80
+    steps:
+      - name: Install base dependencies
+        run: |
+          apt-get update && apt-get install --yes git make python3 python3-pip build-essential
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 10
+      - name: Install node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "22.x"
+          cache: pnpm
+          cache-dependency-path: app/pnpm-lock.yaml
+      - name: Install node dependencies
+        working-directory: app
+        run: pnpm install
+      - name: Install Rust-specific dependencies for sd-proxy
+        run: apt-get install --yes build-essential curl libssl-dev pkg-config
+      - name: Install Rust to build sd-proxy
+        uses: dtolnay/rust-toolchain@1.87.0
```

Comment:
[Elsewhere](https://github.com/freedomofpress/securedrop-client/blob/237e04db72fb2b8e1b907f274a38cdb32cecb3dd/.github/workflows/security.yml#L11-L12) we've left a comment about keeping the Rust version in sync, but let's not worry about that here: I think <https://github.com/freedomofpress/securedrop-dev-docs/issues/245#issuecomment-3079700220> will be a better general approach for Rust *and* Node.js.